### ISSUE TITLE
localize_tso unnecessary calls?

### ISSUE THREAD

**Issue by jbrockmendel** [pandas-dev/pandas#17944]
Related: #17876, #17734 

In `tslib._localize_tso` there are 4 of blocks of the form:

```
        if obj.value != NPY_NAT:
            pandas_datetime_to_datetimestruct(obj.value + delta,
                                              PANDAS_FR_ns, &obj.dts)
        else:
            pandas_datetime_to_datetimestruct(obj.value,
                                              PANDAS_FR_ns, &obj.dts)

```

But in the case where `obj.value == NPY_NAT`, can't we skip the call to `pandas_datetime_to_datetimestruct` because we're going to end up with `NaT` anyway?

**Comment by gfyoung** [comment]
@jbrockmendel : There's one way to find out: remove the calls and run the tests :smile:

**Comment by jbrockmendel** [comment]
Doesn't cause any test failures; kind of hoping for outside confirmation though.

**Comment by jbrockmendel** [comment]
This can be fixed in a follow-up after #17875 is closed
